File: backend/src/main/java/space/davids_digital/sweetie/rest/dto/ChatDto.java
package space.davids_digital.sweetie.rest.dto;

import java.util.List;

public record ChatDto(
        long peerId,
        String title,
        String pictureUrl,
        boolean botEnabled,
        String context,
        long gptMaxInputTokens,
        long gptMaxOutputTokens,
        double gptTemperature,
        double gptTopP,
        double gptFrequencyPenalty,
        double gptPresencePenalty,
        String gptModel,
        boolean processAudioMessages,
        List<String> availableGptModels
) {}

File: backend/src/main/java/space/davids_digital/sweetie/rest/dto/ChatDto.kt
package space.davids_digital.sweetie.rest.dto

data class ChatDto(
    val peerId: Long,
    val title: String?,
    val pictureUrl: String?,
    val botEnabled: Boolean,
    val context: String,
    val gptMaxInputTokens: Long,
    val gptMaxOutputTokens: Long,
    val gptTemperature: Double,
    val gptTopP: Double,
    val gptFrequencyPenalty: Double,
    val gptPresencePenalty: Double,
    val gptModel: String,
    val processAudioMessages: Boolean,
    val availableGptModels: List<String>
)
